core/src/test/scala/kafka/tools/LogCompactionTester.scala

```diff
@@ -360,14 +358,17 @@ object LogCompactionTester {
   def readString(buffer: ByteBuffer): String = {
     val bytes = new Array[Byte](buffer.remaining)
     buffer.get(bytes)
-    new String(bytes, UTF_8)
+    Utils.utf8(bytes)
   }
 
 }
 
 case class TestRecord(topic: String, key: Int, value: Long, delete: Boolean) {
   def this(pieces: Array[String]) = this(pieces(0), pieces(1).toInt, pieces(2).toLong, pieces(3) == "d")
-  def this(line: String) = this(line.split("\t"))
   override def toString = topic + "\t" + key + "\t" + value + "\t" + (if (delete) "d" else "u")
   def topicAndKey = topic + key
 }
+
+object TestRecord {
+  def apply(line: String): TestRecord = new TestRecord(line.split("\t"))
```

Comment:
The idea was to call this method `parse` to make it clear that it's parsing the line and can fail.